TheJumpCloud/support
# Sa 3059 radius cert scripts

## Issues
* [SA-3059](https://jumpcloud.atlassian.net/browse/SA-3059) - Radius User Cert Generation Automation

## What does this solve?

This script provides administrators a way to generate user certificates for passwordless user authentication. 

## Is there anything particularly tricky?

## How should this be tested?

Foreach of the three types of user certs `EmailSAN`, `EmailDN`, `UsernameCn` Generate user certificates for users in a user group. Deploy and test that each of these generated .pfx certificates allow a user to access a radius network.

| Radius Cert  	| MacOS 	| WindowsOS 	|
|--------------	|-------	|-----------	|
| EmailSAN     	| ✅     	| ✅         	|
| EmailDN      	| ✅      	| ✅          	|
| UsernameCn   	| ✅      	| ✅          	|

In testing this morning, I'm finding that both the `EmailDN` and `UsernameCN` certs allow access to my radius network. The `EmailSAN` cert will always generate an error:
 
![Screen Shot 2022-12-15 at 9 26 38 AM](https://user-images.githubusercontent.com/54448601/207914684-0fbb45a5-13cc-4221-bba5-6bc0f3d24fb3.png)

When examining the generated CSR, the subject Alternative Name is recorded. To view the configured attributes run:

`openssl req -text -noout -verify -in ~/Documents/GitHub/support/scripts/automation/Radius/UserCerts/defaultadmin-cert-req.csr`

![Screen Shot 2022-12-15 at 9 28 11 AM](https://user-images.githubusercontent.com/54448601/207915036-479358d9-c956-4a7b-a023-ce1dc2037c22.png)


## Screenshots
